exporters/zipkin/src/test/java/io/opentelemetry/exporters/zipkin/ZipkinSpanExporterEndToEndHttpTest.java

```diff
@@ -0,0 +1,192 @@
+        buildZipkinExporterConfiguration(
+            zipkin.httpUrl() + ENDPOINT_V2_SPANS, Encoding.JSON, SpanBytesEncoder.PROTO3);
+
+    ZipkinSpanExporter zipkinSpanExporter = ZipkinSpanExporter.create(configuration);
+
+    SpanData spanData = buildStandardSpan().build();
+    SpanExporter.ResultCode resultCode = zipkinSpanExporter.export(Collections.singleton(spanData));
+
+    assertThat(resultCode).isEqualTo(SpanExporter.ResultCode.FAILURE);
+    List<Span> zipkinSpans = zipkin.getTrace(TRACE_ID);
+    assertThat(zipkinSpans).isNotNull();
+    assertThat(zipkinSpans).isEmpty();
+  }
+
+  private static ZipkinExporterConfiguration buildZipkinExporterConfiguration(
+      String endpoint, Encoding encoding, SpanBytesEncoder encoder) {
+    return ZipkinExporterConfiguration.builder()
+        .setSender(URLConnectionSender.newBuilder().endpoint(endpoint).encoding(encoding).build())
+        .setServiceName(SERVICE_NAME)
+        .setEncoder(encoder)
+        .build();
+  }
+
+  /**
+   * Exports a span, verify that it was received by Zipkin, and check that the span stored by Zipkin
+   * matches what was sent.
+   */
+  private void exportAndVerify(ZipkinExporterConfiguration configuration) {
+    ZipkinSpanExporter zipkinSpanExporter = ZipkinSpanExporter.create(configuration);
+
+    SpanData spanData = buildStandardSpan().build();
+    SpanExporter.ResultCode resultCode = zipkinSpanExporter.export(Collections.singleton(spanData));
+
+    assertThat(resultCode).isEqualTo(SpanExporter.ResultCode.SUCCESS);
+    List<Span> zipkinSpans = zipkin.getTrace(TRACE_ID);
+
+    assertThat(zipkinSpans).isNotNull();
+    assertThat(zipkinSpans.size()).isEqualTo(1);
+    assertThat(zipkinSpans.get(0)).isEqualTo(buildZipkinSpan());
+  }
+
+  private static SpanData.Builder buildStandardSpan() {
+    return SpanData.newBuilder()
+        .setTraceId(TraceId.fromLowerBase16(TRACE_ID, 0))
+        .setSpanId(SpanId.fromLowerBase16(SPAN_ID, 0))
+        .setParentSpanId(SpanId.fromLowerBase16(PARENT_SPAN_ID, 0))
+        .setTraceFlags(TraceFlags.builder().setIsSampled(true).build())
+        .setStatus(Status.OK)
+        .setKind(Kind.SERVER)
+        .setHasRemoteParent(true)
+        .setName(SPAN_NAME)
+        .setStartEpochNanos(START_EPOCH_NANOS)
+        .setAttributes(attributes)
+        .setTotalAttributeCount(attributes.size())
+        .setTimedEvents(annotations)
+        .setLinks(Collections.<SpanData.Link>emptyList())
+        .setEndEpochNanos(END_EPOCH_NANOS)
+        .setHasEnded(true);
+  }
+
+  private static Span buildZipkinSpan() {
+    return Span.newBuilder()
+        .traceId(TRACE_ID)
+        .parentId(PARENT_SPAN_ID)
+        .id(SPAN_ID)
+        .kind(Span.Kind.SERVER)
+        .name(SPAN_NAME)
+        .timestamp(START_EPOCH_NANOS / 1000)
+        .duration((END_EPOCH_NANOS / 1000) - (START_EPOCH_NANOS / 1000))
+        .localEndpoint(localEndpoint)
+        .addAnnotation(RECEIVED_TIMESTAMP_NANOS / 1000, "RECEIVED")
```

Comment:
looks good. actually the annotations themselves we can revisit, but in any case this E to E makes future discussions easier.